### PR TITLE
fix(deployment): support redeploy state transitions (#3291)

### PR DESCRIPTION
**Pull request by Kroustille**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.
>
> ℹ️ **Feature environment** — A dedicated preview environment (`https://dev-pr-{number}.hub.staging.filigran.io`) is automatically deployed when CI passes. Add the `skip-feature-env` label to opt out.

When a trial instance fails to deploy, the platform may remove it before redeploying it. The Hub polls
platform status roughly once per minute, so short-lived states can be missed entirely.

Until now, `VALID_PLATFORM_STATE_TRANSITIONS` only accepted `removed → provisioning`. If the removal
completed between two polls, the Hub observed `removing → provisioning` and rejected the update with
`DeploymentRequestStatusUpdateNotAllowed`. Worse, `actual_state` then stayed stuck on `removing`, so
**every subsequent status update for that deployment was rejected too** — the deployment never recovered.

This PR makes the two "back to provisioning" transitions valid, conditioned on the deployment request's
stored `target_state` being `active`:

| From        | To             | Condition                |
|-------------|----------------|--------------------------|
| `removing`  | `provisioning` | `target_state = active`  |
| `removed`   | `provisioning` | `target_state = active`  |

`target_state` is not part of `UpdateDeploymentRequestInput`, so it is read from the persisted deployment
request at the single call site (`checkStatusAndDataValidity`). No GraphQL schema change is involved.

https://github.com/user-attachments/assets/d55ffdb9-d4ad-46af-9d37-fd6f0a1ba1ef

https://github.com/user-attachments/assets/ddce2cc9-ab82-4155-ad30-e56652677213

1. Create a trial deployment request with `hub_status = provisioning`, `target_state = active`,
   `actual_state = removing`.
2. Call `updateDeploymentRequest` with `actual_state = provisioning`.
   - **Before:** rejected with `DeploymentRequestStatusUpdateNotAllowed`, deployment stuck on `removing`.
   - **After:** accepted, `actual_state = provisioning` and `hub_status = provisioning`.
3. Repeat with `target_state = removed` (e.g. a cancelled deployment) and verify the update is still rejected.
4. Automated: `cd apps/backend && yarn test src/modules/deployment`

## Related issues

- Related to #3291 

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code
- [x] I added the `skip-feature-env` label if a feature environment is not needed for this PR

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [x] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [x] Integration tests
- [ ] E2E tests
- [ ] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/backend/src/modules/deployment/deployment.helper.test.ts [hunk 1]
```diff
@@ -108,69 +110,43 @@ describe('isHubStatusTransitionValid', () => {
...
+    ${DeploymentRequestPlatformState.Active}       | ${null}                                         | ${DeploymentRequestPlatformState.Active}  | ${'unknown next state'}
+    ${DeploymentRequestPlatformState.Removed}      | ${DeploymentRequestPlatformState.Active}        | ${DeploymentRequestPlatformState.Active}  | ${'removed platform cannot become active directly'}
+    ${DeploymentRequestPlatformState.Removing}     | ${DeploymentRequestPlatformState.Provisioning}  | ${DeploymentRequestPlatformState.Removed} | ${'redeploy is not expected while removal is targeted'}
+    ${DeploymentRequestPlatformState.Removed}      | ${DeploymentRequestPlatformState.Provisioning}  | ${DeploymentRequestPlatformState.Removed} | ${'redeploy is not expected while removal is targeted'}
+    ${DeploymentRequestPlatformState.Removing}     | ${DeploymentRequestPlatformState.Provisioning}  | ${null}                                   | ${'no target state to justify a redeploy'}
+    ${DeploymentRequestPlatformState.Removed}      | ${DeploymentRequestPlatformState.Provisioning}  | ${null}                                   | ${'no target state to justify a redeploy'}
```
Comment (carinelebas) [inline apps/backend/src/modules/deployment/deployment.helper.test.ts]: [nit] I'm not sure these 2 tests really mean something: target_state is in fact never null, so this doesn't reflect a real scenario

**Review by carinelebas — APPROVED** [review]
maybe it would also be worth also fixing copilot comment ?

**Review by carinelebas — APPROVED** [review]
